server/graphql/v2/object/Host.js

```diff
@@ -534,9 +548,68 @@ export const Host = new GraphQLObjectType({
             const collectiveIds = collectives.map(collective => collective.id);
             where.CollectiveId = { [Op.in]: collectiveIds };
           }
+
+          const contributionAmountOverTime = async () => {
+            let contributionAmountOverTime;
+            if (args.timeUnit) {
+              const dateFrom = args.dateFrom ? moment(args.dateFrom).toISOString() : undefined;
+              const dateTo = args.dateTo ? moment(args.dateTo).toISOString() : undefined;
+              contributionAmountOverTime = await sequelize.query(
+                `SELECT DATE_TRUNC(:timeUnit, "createdAt") AS "date", sum(amount) as "amount", "currency"
+                 FROM "Transactions"
+                 WHERE kind = 'CONTRIBUTION'
+                   AND "HostCollectiveId" = :hostCollectiveId
+                   AND type = 'CREDIT'
+                   ${collectiveIds ? `AND "CollectiveId" IN (:collectiveIds)` : ``}
+                   ${dateFrom ? `AND "createdAt" >= :dateFrom` : ``}
+                   ${dateTo ? `AND "createdAt" <= :dateTo` : ``}
+                 GROUP BY DATE_TRUNC(:timeUnit, "createdAt"), "currency"
+                 ORDER BY DATE_TRUNC(:timeUnit, "createdAt")
+                `,
+                {
+                  type: sequelize.QueryTypes.SELECT,
+                  replacements: {
+                    hostCollectiveId: host.id,
+                    timeUnit: args.timeUnit,
+                    collectiveIds,
+                    dateFrom,
+                    dateTo,
+                  },
+                },
+              );
+              contributionAmountOverTime = contributionAmountOverTime.map(contributionAmount =>
+                convertCurrencyAmount(
+                  contributionAmount.currency,
+                  host.currency,
+                  contributionAmount.date,
+                  contributionAmount.amount,
+                ),
+              );
+              contributionAmountOverTime = await Promise.all(contributionAmountOverTime);
+
+              const counts = contributionAmountOverTime.reduce((prev, curr) => {
+                const count = prev.get(curr.date) || 0;
+                prev.set(curr.date.toISOString(), curr.amount + count);
+                return prev;
+              }, new Map());
+
+              contributionAmountOverTime = [...counts].map(([date, amount]) => {
+                return { date, amount, currency: host.currency };
+              });
```

Comment:
I don't understand why we would want to have `amountForThisMonth = SUM(previousMonthsAmounts)`. And even if we wanted that, the resulting metric would be false here since we're not fetching the entire history.

Reply:
Yes, seems this is a mistake on my part. We don't need this sum. I've corrected that.